sql/compression_defaults.sql

```diff
@@ -26,38 +26,55 @@ BEGIN
 
     SELECT * INTO STRICT _hypertable_row FROM _timescaledb_catalog.hypertable h WHERE h.table_name = _table_name AND h.schema_name = _schema_name;
 
-    --STEP 1 if column stats exist use unique indexes. Pick the column that comes first in any such indexes. Ties are broken arbitrarily.
+    --STEP 1 if column stats exist use unique indexes.
+    --Pick the column that comes first in any such indexes. Select the column with the most even distribution.
     --Note: this will only pick a column that is NOT unique in a multi-column unique index.
     with index_attr as (
-        SELECT
+      SELECT
         a.attnum, min(a.pos) as pos
-        FROM
-            (select indkey, indnkeyatts from pg_catalog.pg_index where indisunique and indrelid = relation) i
-        INNER JOIN LATERAL
-            (select * from unnest(i.indkey) with ordinality) a(attnum, pos) ON (TRUE)
-        WHERE a.pos <= i.indnkeyatts
-        GROUP BY 1
-    )
-    SELECT
-      a.attname INTO _segmentby
-    FROM
-      index_attr i
-    INNER JOIN
-      pg_attribute a on (a.attnum = i.attnum AND a.attrelid = relation)
-    --right now stats are from the hypertable itself. Use chunks in the future.
-    INNER JOIN pg_stats s ON (s.attname = a.attname and s.schemaname = _schema_name and s.tablename = _table_name)
-    WHERE
-      a.attname NOT IN (SELECT column_name FROM _timescaledb_catalog.dimension d WHERE d.hypertable_id = _hypertable_row.id)
+      FROM (
+        SELECT indkey, indnkeyatts
+        FROM pg_catalog.pg_index
+        WHERE indisunique AND indrelid = relation
+      ) i
+      INNER JOIN LATERAL (
+        SELECT * FROM unnest(i.indkey) WITH ORDINALITY
+      ) a(attnum, pos) ON TRUE
+      WHERE a.pos <= i.indnkeyatts
+      GROUP BY a.attnum
+    ),
+    stats_with_stddev as (
+      SELECT
+        a.attname,
+        i.pos,
+        ROUND(stddev_pop(freqs)::numeric, 5) as freq_stddev
+      FROM index_attr i
+      INNER JOIN pg_attribute a ON a.attnum = i.attnum AND a.attrelid = relation
+      INNER JOIN pg_stats s ON s.attname = a.attname
+                            AND s.schemaname = _schema_name
+                            AND s.tablename = _table_name
```

Comment:
```suggestion
                            AND s.tablename = _table_name
                            AND s.inherited = true
```
pg_statistics primary key is relid, attnum, inherit so there might be an inherited and non-inherited entry and we want the inherited entry